gprofiler/main.py

```diff
@@ -136,13 +135,14 @@ def __init__(
         # 2. accessible only by us.
         # the latter can be root only. the former can not. we should do this separation so we don't expose
         # files unnecessarily.
-        self._temp_storage_dir = TemporaryDirectoryWithMode(dir=TEMPORARY_STORAGE_PATH, mode=0o755)
-        self.system_profiler, self.process_profilers = get_profilers(
-            user_args,
-            storage_dir=self._temp_storage_dir.name,
-            stop_event=self._stop_event,
-            profile_spawned_processes=self._profile_spawned_processes,
+        self._profiler_state = ProfilerState(
+            Event(),
```

Comment:
Note that there is still a `GProfiler._stop_event` initialized and it's initialized to a DIFFERENT `Event()` object. Please use the same object as well - i.e, remove `self._stop_event` and use `self._profiler_state.stop_event`.